lib/rack/request.rb

```diff
@@ -598,35 +598,29 @@ def split_header(value)
         value ? value.strip.split(/[,\s]+/) : []
       end
 
-      AUTHORITY = /^
-        # The host:
+      AUTHORITY = /
+        \A
```

Comment:
@pvande what is the difference between `\A` and `\Z` vs `^` and `$`?

Reply:
`\A` matches the beginning of the string.  `^` matches any beginning of line in the string. `\Z` matches the end of string, allowing possible newline at the end of the string. `$` matches the end of any line in the string.

In general, you are much more likely to want `\A` and `\Z` (or `\z`, which matches the end of string) than `^` and `$`.  Using `^` and `$` usually leads to bugs, in my experience.